mobile/src/people/departments/departments-horizontal-scrollable-list.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import { Animated, Easing, View, Text, ScrollView, Dimensions, TouchableOpacity, ScrollResponderEvent } from 'react-native';
+import { AppState } from '../../reducers/app.reducer';
+import { connect, Dispatch, MapStateToProps, MapDispatchToPropsFunction } from 'react-redux';
+import { EmployeeCardWithAvatar } from '../employee-card-with-avatar';
+import { DepartmentsTree } from './departments-tree';
+import { DepartmentsTreeNode } from './departments-tree-node';
+import { Employee } from '../../reducers/organization/employee.model';
+import { PeopleActions, updateDepartmentIdsTree } from '../../reducers/people/people.action';
+import { loadEmployeesForDepartment } from '../../reducers/organization/organization.action';
+
+interface DepartmentsHScrollableListProps {
+    departmentsTree: DepartmentsTree;
+    treeLevel?: number;
+    departmentsTreeNodes?: DepartmentsTreeNode[];
+    headDepartment: DepartmentsTreeNode;
+    topOffset?: number;
+}
+
+interface DepartmentsHScrollableListDispatchProps {
+    requestEmployeesForDepartment: (departmentId: string) => void;
+    updateDepartmentIdsTree: (index: number, departmentId: string) => void;
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<PeopleActions>) => ({
```

Comment:
I would pass dispatch actions through the props from the people-company component, it would get rid of connect function

Reply:
addressed